kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/AssetMaintainableImpl.java

```diff
@@ -0,0 +1,111 @@
+package edu.arizona.kfs.module.cam.document;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.kns.document.MaintenanceDocument;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.service.KRADServiceLocator;
+import org.kuali.rice.krad.service.NoteService;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.sys.KFSConstants;
+
+@SuppressWarnings("deprecation")
+public class AssetMaintainableImpl extends org.kuali.kfs.module.cam.document.AssetMaintainableImpl {
+
+    private static final long serialVersionUID = 5710219294064590806L;
+
+    @Override
+    public void processAfterEdit(MaintenanceDocument document, Map<String, String[]> parameters) {
+        List<Note> notes = new ArrayList<Note>();
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.CHANGE);
+        document.setNotes(notes);
+
+        super.processAfterEdit(document, parameters);
+    }
+
+    @Override
+    public void setGenerateDefaultValues(String docTypeName) {
+        super.setGenerateDefaultValues(docTypeName);
+
+        List<Note> notes = new ArrayList<Note>();
+
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+
+            if (notes.isEmpty()) {
+                Note newBONote = getNewBoNoteForAdding(KFSConstants.CreateAndUpdateNotePrefixes.ADD);
+                notes.add(newBONote);
+            }
+        }
+    }
+
+    @Override
+    public void setupNewFromExisting(MaintenanceDocument document, Map<String, String[]> parameters) {
+        super.setupNewFromExisting(document, parameters);
+
+        List<Note> notes = new ArrayList<Note>();
+
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.ADD);
+
+        document.setNotes(notes);
```

Comment:
I think that's pretty bold to claim, i.e. that the notes are not used outside that last method -- why else persist to the DB in the former if it's never to be used? This would force anyone downstream to either: manually load the note themselves, reload the entire document to get cascading refresh on that collection, or plain think that the note saving didn't happen.